src/rotten-tomatoes/SpoilTomatoes.js

```diff
@@ -0,0 +1,71 @@
+class SpoilTomatoes {
+
+    constructor(grid) {
+        this.grid = grid;
+        this.colLength = 0;
+        this.rowLength = 0;
+        this.getAdjacents = this.getAdjacents.bind(this)
+    };
+
+    getRottenTomatoes() {
+        const locationOfSpoiltTomatoes = [];
+        let numGoodTomatoe = 0;
+        for (let row = 0; row < this.grid.length; row++) {
+            this.rowLength = this.grid.length - 1;
+            for (let col = 0; col < this.grid[row].length; col++) {
+                this.colLength = this.grid[row].length - 1;
+                if (this.grid[row][col] === 2) {
+                    locationOfSpoiltTomatoes.push([row, col])
+                }
+                if (this.grid[row][col] === 1) {
+                    numGoodTomatoe++
+                }
+            }
+        }
+        return { numGoodTomatoe, locationOfSpoiltTomatoes };
+    }
+
+    getAdjacents(position) {
+        const [row, column] = position;
+        const topAdjacent = row === 0 ? false : [row - 1, column];
+        const rightAdjacent = column === this.colLength ? false : [row, column + 1];
+        const bottomAdjacent = row === this.rowLength ? false : [row + 1, column];
+        const leftAdjacent = column === 0 ? false : [row, column - 1]
+        return [topAdjacent, rightAdjacent, bottomAdjacent, leftAdjacent]
+            .filter(Boolean)
+            .filter(([row, col]) => this.grid[row][col] === 1)
+    }
+
+    getTimeOfSpread(spoiltTomatoesLocations, numGoodTomatoe) {
```

Comment:
I don't think this needs to take arguments since it's not really a generalised solution, it's specific to the grid the class was initialized with.

So

```js
const spoiltTomatoes = new SpoilTomatoes(grid)
return spoiltTomatoes.getTimeOfSpread()
```

would be a better API.

Reply:
Okay, allowing the class handle all the logic while keeping them hidden. Thanks sir